PocketKit/Sources/Sync/NetworkPathMonitor.swift

```diff
@@ -0,0 +1,39 @@
+import Network
+
+
+public protocol NetworkPath {
+    var status: NWPath.Status { get }
+}
+
+public protocol NetworkPathMonitor: AnyObject {
+    typealias UpdateHandler = (NetworkPath) -> Void
+
+    var currentNetworkPath: NetworkPath { get }
+    var updateHandler: UpdateHandler? { get set }
+
+    func start(queue: DispatchQueue)
+}
+
+extension NWPath: NetworkPath {
+
+}
+
+extension NWPathMonitor: NetworkPathMonitor {
+    public var currentNetworkPath: NetworkPath {
+        currentPath
+    }
+
+    public var updateHandler: UpdateHandler? {
+        get {
+            return { (path: NetworkPath) in
+                guard let nwpath = path as? NWPath else {
+                    fatalError("Attempt to use \(path) as path for real \(Self.self). This is not allowed")
+                }
+
+                self.pathUpdateHandler?(nwpath)
+            }
```

Comment:
Because of the way the protocols are set up for mocking, this is _theoretically_ possible. But in reality, we should never hit this since the system always calls the update handler with a proper `NWPath`